qiime2/plugin/plugin.py

```diff
@@ -67,9 +69,10 @@ def __init__(self, name, version, website, package, citation_text=None,
 
         self.formats = {}
         self.views = {}
-        self.types = {}
+        self.type_fragments = {}
         self.transformers = {}
         self.type_formats = []
+        self.types = set()
```

Comment:
This is a set, because a type may be registered to more than one format, but these types should not be duplicated in our collection?

Reply:
I believe that is the idea behind it being a set yes.

Reply:
Formats don't factor in here. I believe we suggested a set here to indicate that there is no order to the collection, we are less concerned with uniqueness.